### PR TITLE
task18 | added solution

### PR DESCRIPTION
**Pull request by TatsiaA**
task18 | added solution

### CODE REVIEW

File: src/Task18/task18.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * @param array $arr
+ * @return int|string
+ */
+function getMajorityNumberInArray(array $arr): int | string
+{
+
+    $set = [];
+    for ($i = 0; $i < count($arr); $i++) {
```
Comment (Jampire) [inline src/Task18/task18.php:15]: `$i < count($arr);` is evaluated on every iteration, but array count is permanent value here, no need to calculate it on every step. Create a variable for it.

File: src/Task18/task18.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        if (array_key_exists($arr[$i], $set)) {
+            $count = $set[$arr[$i]] + 1;
+            if ($count > count($arr) / 2) {
```
Comment (Jampire) [inline src/Task18/task18.php:18]: Once again, array count is permanent value here.

File: src/Task18/task18.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                return $arr[$i];
+            } else {
+                $set[$arr[$i]] = $count;
+            }
+        } else {
+            $set[$arr[$i]] = 1;
+        }
+    }
+
+    return "No Majority element";
```
Comment (Jampire) [inline src/Task18/task18.php:28]: This sentence can be actual majority element as well. For example:
```php
[
    'No Majority element',
    'No Majority element',
    3,
    'No Majority element',
];
```

File: tests/Task18/Task18test.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+     */
+    public function testGetMajorityNumberInArray(array $arr, int | string $res): void
+    {
+        $this->assertEquals($res, getMajorityNumberInArray($arr));
+    }
+
+    public function dataProvider(): array
+    {
+        return [
+            '5 is majority element' => [[1, 2, 3, 4, 5, 6, 5, 5, 5, 5, 5, 6, 5], 5],
```
Comment (Jampire) [inline tests/Task18/Task18test.php:25]: You forgot to test edge cases.
```php
'empty string is a majority element' => ['', ['','','']],
'null is a majority element' => [null, [null, null, null]],
'3 is a majority element' => [3, [null, 3, 3]],
'b is a majority element' => ['b', ['a' => 'b']],
```

**Review by Jampire — CHANGES_REQUESTED** [review]

File: src/Task18/task18.php (hunk 3 above)
Comment (TatsiaA) [reply src/Task18/task18.php]: thrown exception

File: tests/Task18/Task18test.php (hunk 4 above)
Comment (TatsiaA) [reply tests/Task18/Task18test.php]: changed solution
